### PR TITLE
fix weakhashmap

### PR DESCRIPTION
**Pull request by DrRacket**
Rewrite the logic for weakhashmap. Fixed the remove bug.
now the weakhashmap shuffling logic does not use hashiteratorshuffler.

### CODE REVIEW

File: nondex-core/src/main/java/java/util/WeakHashMap.java [hunk 1]
```diff
@@ -763,6 +763,8 @@ public String toString() {
         private Entry<K,V> lastReturned = null;
         private int expectedModCount = modCount;
 
+//        HashIteratorShuffler shuffler;
```
Comment (alexgyori) [inline nondex-core/src/main/java/java/util/WeakHashMap.java]: Why this comment?

File: nondex-core/src/main/java/java/util/WeakHashMap.java [hunk 2]
```diff
@@ -775,13 +777,33 @@ public String toString() {
...
+        public Entry<K, V> nextEntry() {
+            if (modCount != expectedModCount)
+                throw new ConcurrentModificationException();
+            lastReturned = iter.next();
+            currentKey = lastReturned.get();
+            return lastReturned;
+        }
+
+        public boolean originalHasNext() {
+            Entry<K,V>[] t = table;
```
Comment (alexgyori) [inline nondex-core/src/main/java/java/util/WeakHashMap.java]: You should not need to edit the originalHasNext. Why do you need this?

File: nondex-core/src/main/java/java/util/WeakHashMap.java (hunk 1 above)
Comment (DrRacket) [inline nondex-core/src/main/java/java/util/WeakHashMap.java]: Will delete this.

File: nondex-core/src/main/java/java/util/WeakHashMap.java (hunk 2 above)
Comment (DrRacket) [inline nondex-core/src/main/java/java/util/WeakHashMap.java]: I actually didn't modify it. You can look at line 783 and line 806 and you will find that this is because of how the "diff" works.

File: nondex-test/src/test/java/edu/illinois/nondex/core/IdentityHashMapTest.java [hunk 3]
```diff
@@ -1,53 +0,0 @@
-/*
```
Comment (alexgyori) [inline nondex-test/src/test/java/edu/illinois/nondex/core/IdentityHashMapTest.java]: why did you remove this here? Seems unrelated to the PR.

File: nondex-core/src/main/java/java/util/WeakHashMap.java [hunk 4]
```diff
@@ -763,6 +763,7 @@ public String toString() {
         private Entry<K,V> lastReturned = null;
         private int expectedModCount = modCount;
 
+
```
Comment (DrRacket) [inline nondex-core/src/main/java/java/util/WeakHashMap.java]: Will delete.

File: nondex-test/src/test/java/edu/illinois/nondex/core/IdentityHashMapTest.java (hunk 3 above)
Comment (DrRacket) [inline nondex-test/src/test/java/edu/illinois/nondex/core/IdentityHashMapTest.java]: Yeah.. Should be in another PR.. I will discard the change.
